### DISCUSSION TITLE
Problem with yi-hack_ha_integration and diferent yi-home cameras models with yi-hack

### DISCUSSION THREAD

**Discussion by golvellius** [roleoroleo/yi-hack_ha_integration#141] [General]
Hi, 

First of all i would like to thank you for your supperb work to integrate yi-home cameras ( new firmwares and ha integrations). 

I have been playing with different models of cameras .. i have 3 ( one yi-hack-v5 another Mstar and one allwinner V1)..
In all cameras i have the same problem when integrated in HA, Status connection always disconnected .. 

Example if i switch off led light it does not switch off .. In the webpage is working .. but in the HA integration control no. 
The only sensor working is when i press the link button , i can see the rstp stream ...  but i can not enable detections .. etc .. 
You can check photos..

![Ha_Inte1](https://github.com/roleoroleo/yi-hack_ha_integration/assets/17733658/498225d3-47b9-4cd1-96d3-b8dbdde956ae)

![Ha_yi1](https://github.com/roleoroleo/yi-hack_ha_integration/assets/17733658/5e646b9e-91cc-47b3-941e-32cb506b56c6)

Now i am playing with the allwinner model (y20ga). I know that for the V5 models are limited .. But for the Mstar model Y25 i have got the same problems .. All 3 types of cameras the same problem. I have repeated to get default values and restart the cameras but the same problem .. Also i have checked that in the camera all MQTT data are disabled... .. Tested different versions of the integration .. also installed manually from your github repository ... But no luck .. 

Some info:
HA Core : 2023.12.4
Supervisor: 2023.12.1.dev2104
Frontend: 20231208.2
Running on a orangepi 2 arm64 .. 

MODEL DATA:
Firmware Version | 0.3.7
Base Version | 12.2.0.2_202212191129
Model Suffix | y20ga
Serial Number | 0000BFUSxxxxxxxxxxxx
Local Time | Thu Jan 4 02:32:39 CET 2024
Uptime | 1h 18m 5s
Load Average | 1.28 0.99 0.90
Free/Total Memory | 10592/61164 KB
Free SD Space | 99%

If you need more information i can sent it to you .. it's not a problem for me to send more. 

So control are not working ... and sensors are not chaning only the one to see the stream...
Please could you help me.?

Thanks for all ... Sorry for the message...

**Comment by golvellius** [comment]
Hi .. 
After investigating and reading some comments .. i think that the problem is solved if i enable the mqtt service on each camera and different client-id and topics prefix from each camera .. now i can enable disable the motion control .. and also all cameras are online in the status tab .. 

Is that correct ? 

Thanks for all and sorry for the other message...

**Reply by roleoroleo** [reply]
I can confirm.
MQTT is required to use the integration and you need to define a unique client id for each cam.

**Comment by AndyB2067** [comment]
I have gone through the process of setting this up in HA but when it comes to adding the integration it does not want to connect. I add the IP address, Port i leave as 80 and have setup port forwarding on the router to 80, username and password entered and the rest of the setup as is. 
![image](https://github.com/roleoroleo/yi-hack_ha_integration/assets/173922744/6b26a9a7-8b48-4d1c-b81e-51671a9316f4)
what if anything obvious am i missing?

**Reply by roleoroleo** [reply]
Are ha and the cam in the same lan?

**Comment by AndyB2067** [comment]
The cam and HA are on the same Lan. HA is set up on the PC as a virtual machine.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by roleoroleo** [comment]
It should work if you configure properly the network.
Check the NAT configuration.

**Comment by AndyB2067** [comment]
Hi Mate not sure how to do that.
________________________________

**Comment by AndyB2067** [comment]
Hi Roleo,
      this what i get when i enter details, is this correct?
[cid:78d1484b-3696-406d-88e6-de0dfde48873]
When i enter details it goes back to this.
[cid:f235515d-4f8d-43ec-a2cd-39ddadfc40a6]
Thanks Andy
________________________________

**Comment by roleoroleo** [comment]
What's the ip address of the virtual machine where ha is installed?
What's the ip address of the cam?
Are you able to ping ha from the cam and vice-versa?

**Comment by AndyB2067** [comment]
Hi Roleo,
      What's the ip address of the virtual machine where ha is installed? 192.168.1.109
      What's the ip address of the cam? 192.168.1.123
      Are you able to ping ha from the cam and vice-versa? I don't know how to do this.

I appreciate all of the help you give.
Andy


________________________________

**Comment by AndyB2067** [comment]
Hi,
      So i have gone into HA and the then Terminal. Typed in ping and the IP and it seems to be pinging.
[cid:9d838507-dd08-4777-bf76-59bf2aa14260]
 i don't however know how to do this the other way.
________________________________

**Comment by roleoroleo** [comment]
Your configuration is quite simple.
It should work.
Have you set a password in the cam?
![image](https://github.com/user-attachments/assets/feb9340e-e198-4294-b63c-6422f8f41ff9)

How do you fill the field "Host"?

**Comment by AndyB2067** [comment]
I do have a password and use it correctly.
Host i have tried https://192.168.1.123<https://192.168.1.123/>, http://192.168.1.123<http://192.168.1.123/> and 192.168.1.123
________________________________

**Comment by roleoroleo** [comment]
Use just the IP address: 192.168.1.123
It must works.

**Comment by AndyB2067** [comment]
Would the port forwarding have any bearing on the issue?
[cid:cf6b6a5c-d5ce-438c-b3b7-3f0c8d2321ea]
________________________________

**Comment by roleoroleo** [comment]
The cam and ha are in the same network, port forwarding is not a problem.
Are you able to read the log?
Menu Settings -> System -> Logs
Is there any message about yi-hack when you try to add a cam?

**Comment by AndyB2067** [comment]
Yep i cot this error log

This error originated from a custom integration.

Logger: custom_components.yi_hack.common
Source: custom_components/yi_hack/common.py:45
integration: Yi Cam with yi-hack (documentation, issues)
First occurred: 17:45:15 (1 occurrences)
Last logged: 17:45:15

Failed to get status from device 192.168.1.123: error HTTPConnectionPool(host='192.168.1.123', port=8080): Max retries exceeded with url: /cgi-bin/status.json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fbf40c07bc0>: Failed to establish a new connection: [Errno 111] Connection refused'))

Not sure about this bit
Dependencies

  1.  Home Assistant
     *   MQTT Integration<https://www.home-assistant.io/integrations/mqtt> installed
     *
Mosquitto broker<https://github.com/home-assistant/addons/tree/master/mosquitto> add-on installed and connected to MQTT  These are installed
  2.  Camera
     *   MQTT enabled
     *   Configured with MQTT Broker credentials   Not sure about this section
     *   Default Topics configuration

________________________________

**Comment by roleoroleo** [comment]
HA is not able to reach the cam.
But yes... MQTT is required to run the integration.
Install it in HA.

**Comment by AndyB2067** [comment]
It is install just not configured.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by roleoroleo** [comment]
Check if it's working.
You can use MQTT Explorer (Windows).

**Comment by AndyB2067** [comment]
Cannot find in Windows and when you go to MQTT Explorer via internet it points https://mqtt-explorer.com/. When i click to download page says [cid:37b8b609-3eb3-48ed-ac8f-52444a52536c]
MQTT Explorer<https://mqtt-explorer.com/>
An all-round MQTT client that provides a structured topic overview
mqtt-explorer.com

________________________________

**Comment by roleoroleo** [comment]
Download link:
https://github.com/thomasnordquist/MQTT-Explorer/releases/download/v0.4.0-beta.6/MQTT-Explorer-Setup-0.4.0-beta.6.exe
